spec/system/admin/payments_stripe_spec.rb

```diff
@@ -19,8 +19,9 @@
   end
 
   before { setup_stripe }
+  after { clear_stripe }
```

Comment:
I just had another idea which would prevent us from forgetting to clear Stripe:
```suggestion
  around do |example|
    with_stripe_setup { example.run }
  end
```

And the helper something like:
```rb
def with_stripe_setup(api_key = "sk_test_12345", publishable_key = "pk_test_12345")
  original_keys = [Stripe.api_key, Stripe.publishable_key]

  Stripe.api_key = api_key
  Stripe.publishable_key = publishable_key
  allow(Spree::Config).to receive(:stripe_connect_enabled).and_return(true)

  yield

  Stripe.api_key, Stripe.publishable_key = original_keys
end
```

This method makes sure that we always restore the original state, whatever it is. We could even nest calls.

Reply:
Thanks for that suggestion @mkllnk  I totally agree it would be much better to have this within `around` block - I had tried it before with no success, but we're getting closer with your code now. It is throwing the error:
`       The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported.`

So I did some digging and it seems we need to wrap
`allow(Spree::Config).to receive(:stripe_connect_enabled).and_return(true)`

around `RSpec::Mocks.with_temporary_scope`; for reference, taken from [this discussion](https://github.com/rspec/rspec-core/issues/1598#issuecomment-351794624).

After this change your suggestion works like a charm, and we need not remember to clear stripe settings. I'll commit and make the changes on the necessary files.

Thanks!

Reply:
Ok, so using `RSpec::Mocks.with_temporary_scope` seems not to really enable Stripe. It seems this is not always necessary and indeed, using `allow(Spree::Config).to receive(...` already prevents leaking. So perhaps it is a better idea to include only the api and private keys within the `around` block, since these are the settings leaking and causing issues.

Enabling/disabling Stripe is now included as helpers and can be called independently of the `around` block.